### PR TITLE
Remove all jbuild parsers

### PR DESCRIPTION
**Pull request by rgrinberg**
This also requires porting some old tests to dune.

### CODE REVIEW

File: test/blackbox-tests/test-cases/dune-ppx-driver-system/run.t [hunk 1]
```diff
@@ -175,26 +175,10 @@ Test using installed drivers
...
-                     ^^^^^^^^^^^^^^^^^^
-  Error: Rule failed to generate the following targets:
-  - test_ppx_args.pp.ml
+  File "jbuild", line 2, characters 2-19:
+  2 |  ((name testdriver)
+        ^^^^^^^^^^^^^^^^^
+  Error: Atom expected
   [1]
 
   $ dune build --display short --root jbuild-driver @install
```
Comment (diml) [inline test/blackbox-tests/test-cases/dune-ppx-driver-system/run.t]: Should we just delete this altogether?

File: test/blackbox-tests/test-cases/too-many-parens/run.t [hunk 2]
```diff
@@ -3,61 +3,43 @@ are readable.
 
   $ dune build --root a
```
Comment (diml) [inline test/blackbox-tests/test-cases/too-many-parens/run.t]: We can probably delete this whole test as well

File: test/blackbox-tests/test-cases/upgrader/run.t [hunk 3]
```diff
@@ -16,16 +16,13 @@
   ;
   
   (rule
-   (deps
-    (:< x)
-    y
-    z) ; abc
+   (deps x y z) ; abc
```
Comment (diml) [inline test/blackbox-tests/test-cases/upgrader/run.t]: That doesn't seem right

File: test/blackbox-tests/test-cases/too-many-parens/run.t (hunk 2 above)
Comment (rgrinberg) [reply test/blackbox-tests/test-cases/too-many-parens/run.t]: Deleted

File: test/blackbox-tests/test-cases/dune-ppx-driver-system/run.t (hunk 1 above)
Comment (rgrinberg) [reply test/blackbox-tests/test-cases/dune-ppx-driver-system/run.t]: Deleted

**Comment by diml** [comment]
I get a really weird error when I do `make test`:

```
./_boot/default/bin/main.exe runtest
File "src/dune_project.ml", line 432, characters 68-68:
432 |             parse (enter p) Univ_map.empty (List (Loc.of_pos __POS__, []))
                                                                          
Error: Premature end of list
make: *** [test] Error 1
```

It looks like dune is trying to parse `dune_project.ml` as a `dune` file :confused:

**Comment by emillon** [comment]
I've seen that one before when splitting a module from `dune_file.ml`. I think the bootstrap phase got really confused.

**Comment by diml** [comment]
Oh, the error is actually pointing to `__POS__`... It's because I removed `Stanza.Decoder`, where `list` was redefined as `list t = repeat t` rather than `list t = enter (repeat t)`...

**Comment by diml** [comment]
To avoid confusion, I ended up removing `Decoder.record` and `Decoder.list`. Now we have to use `enter` explicitly if we want to force extra parentheses.

**Pull request merged**